src/test/java/uk/gov/hmcts/reform/prl/services/acro/CsvWriterTest.java

```diff
@@ -0,0 +1,326 @@
+        "Is Applicant Address Confidential", "Is Applicant Phone Confidential", "Is Applicant Email Confidential", "Order File Name"
+    };
+
+    private static final String APPLICANT_PHONE = "1234567890";
+    private static final String APPLICANT_EMAIL = "test@test.com";
+    private static final String APPLICANT_ADDRESS_LINE1 = "123 Example Street";
+    private static final String APPLICANT_ADDRESS_LINE2 = "London";
+    private static final String APPLICANT_POSTCODE = "E1 6AN";
+    private static final String RESPONDENT_ADDRESS_LINE1 = "70 Petty France";
+    private static final String RESPONDENT_ADDRESS_LINE2 = "Westminster";
+    private static final String RESPONDENT_POSTCODE = "SW1H 9EX";
+    private static final String RESPONDENT_PHONE = "07700900123";
+    private static final String RESPONDENT_EMAIL = "respondent@example.com";
+
+    @InjectMocks
+    private CsvWriter csvWriter;
+
+    @BeforeEach
+    void setUp() throws IOException {
+        ReflectionTestUtils.setField(csvWriter, "sourceDirectory", Files.createTempDirectory("csvwriter-test").toString());
+    }
+
+    @Nested
+    @DisplayName("CSV File Creation Tests")
+    class CsvFileCreationTests {
+
+        @Test
+        @DisplayName("Should create CSV file with headers")
+        void shouldCreateCsvFileWithHeaders() throws IOException {
+            File csvFile = csvWriter.createCsvFileWithHeaders();
+
+            assertAll(
+                    "CSV file with headers validation",
+                    () -> assertNotNull(csvFile, "CSV file should not be null"),
+                    () -> assertTrue(csvFile.exists(), "CSV file should exist"),
+                    () -> assertTrue(csvFile.getName().startsWith("manifest-"), "File should start with 'manifest-'"),
+                    () -> assertTrue(csvFile.getName().endsWith(".csv"), "File should have .csv extension")
+            );
+
+            List<String> lines = Files.readAllLines(csvFile.toPath());
+            assertEquals(1, lines.size(), "CSV should have only header line");
+
+            String[] actualHeaders = lines.get(0).split(",");
+            assertArrayEquals(EXPECTED_CSV_HEADERS, actualHeaders, "Headers should match expected format");
+        }
+
+        @Test
+        @DisplayName("Should generate correct filename format")
+        void shouldGenerateCorrectFilenameFormat() throws IOException {
+            File csvFile = csvWriter.createCsvFileWithHeaders();
+
+            String expectedDateFormat = LocalDate.now().format(DateTimeFormatter.ofPattern("yyyyMMdd"));
+            String expectedFilename = "manifest-" + expectedDateFormat + ".csv";
+
+            assertEquals(expectedFilename, csvFile.getName(), "Filename should match expected format");
+        }
+    }
+
+    @Nested
+    @DisplayName("CSV Row Data Creation Tests")
+    class CsvRowDataCreationTests {
```

Comment:
we dont normally split up a Test class with this inner-class grouping
if you find that helps then OK, but then maybe we should have split the class into each individual responsibilities?